metrics/sliding_window_sample.go

```diff
@@ -0,0 +1,133 @@
+package metrics
+
+// go version of sliding window reservoir
+// refer to : https://github.com/dropwizard/metrics/blob/3.2-development/metrics-core/src/main/java/com/codahale/metrics/SlidingWindowReservoir.java
+// and : https://github.com/rcrowley/go-metrics/blob/master/sample.go
+import (
+	"sync"
+
+	gomet "github.com/rcrowley/go-metrics"
+)
+
+// SlidingWindowSample : http://metrics.dropwizard.io/3.1.0/manual/core/#sliding-window-reservoirs
+type SlidingWindowSample struct {
```

Comment:
SlidingWindowSample is not required for now, it just a spare, it gives a limited bucket size, and drop data from first if bucket is full.
SlidingWindowTimeSample is like SlidingWindowSample, it drop data if data is outdated.
Currently, i'm using SlidingWindowTimeSample for statistics latencies and throughput